api/src/main/java/com/google/sps/EmailType.java

```diff
@@ -0,0 +1,19 @@
+package com.google.sps;
+
+import com.google.api.gax.rpc.InvalidArgumentException;
+
+enum EmailType {
+  START, TURN;
+
+  public EmailCreation createEmail(String gameID, String playerEmail, String playerName)
+    throws InvalidArgumentException {
+
+    if(EmailType.this == START) {
+      return EmailCreation.startGameEmail(gameID, new User(playerEmail, playerName));
+    } else if(EmailType.this == TURN) {
+      return EmailCreation.endGameEmail(gameID, new User(playerEmail, playerName));
+    }
+
+    throw new IllegalArgumentException();
```

Comment:
Wait is there a difference between the InvalidArgumentException declared in the method header vs. the IllegalArgumentException thrown on line 17?

Reply:
That needs to be changed, my mistake